### PR TITLE
GEOMESA-3048 Add readiness check for Kafka DataStore Initial Loads

### PR DESCRIPTION
**Pull request by jnh5y**
Signed-off-by: Jim Hughes <jhughes@ccri.com>

### CODE REVIEW

File: geomesa-gs-kafka-catalog-listener/src/main/scala/org/geomesa/gs/rest/kafka/KafkaLoaderStatusController.scala [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import org.springframework.http.{HttpStatus, MediaType}
+import org.springframework.web.bind.annotation.{GetMapping, RequestMapping, RestController}
+
+@RestController
+@RequestMapping(path = Array(RestBaseController.ROOT_PATH + "/kafka"),  produces = Array(MediaType.APPLICATION_JSON_VALUE))
+class KafkaLoaderStatusController extends RestBaseController with LazyLogging {
+  @GetMapping
+  def statusGet(): String = {
+    if (KafkaCacheLoader.LoaderStatus.allLoaded()) {
+      "ok"
```
Comment (elahrvivaz) [inline geomesa-gs-kafka-catalog-listener/src/main/scala/org/geomesa/gs/rest/kafka/KafkaLoaderStatusController.scala:23]: i assume this string doesn't matter, and it's really just the 200 return status we care about?

File: geomesa-gs-kafka-catalog-listener/src/main/scala/org/geomesa/gs/rest/kafka/KafkaLoaderStatusController.scala [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    } else {
+      throw new RestException("KDS Layers still performing initial load.", HttpStatus.SERVICE_UNAVAILABLE)
```
Comment (elahrvivaz) [inline geomesa-gs-kafka-catalog-listener/src/main/scala/org/geomesa/gs/rest/kafka/KafkaLoaderStatusController.scala:25]: similarly, this just serves to return a non-200 code?

**Review by elahrvivaz — APPROVED** [review]

File: geomesa-gs-kafka-catalog-listener/src/main/scala/org/geomesa/gs/rest/kafka/KafkaLoaderStatusController.scala (hunk 2 above)
Comment (jnh5y) [reply geomesa-gs-kafka-catalog-listener/src/main/scala/org/geomesa/gs/rest/kafka/KafkaLoaderStatusController.scala:25]: Yes.  I'll admi that I didn't look too close about how to return a 503 without throwing an exception like this.

File: geomesa-gs-kafka-catalog-listener/src/main/scala/org/geomesa/gs/rest/kafka/KafkaLoaderStatusController.scala (hunk 1 above)
Comment (jnh5y) [reply geomesa-gs-kafka-catalog-listener/src/main/scala/org/geomesa/gs/rest/kafka/KafkaLoaderStatusController.scala:23]: Correct.

**Pull request merged**